File: mobile/lib/models/invoice.dart
// models/invoice.dart
class InvoiceItem {
  final String sku;
  final String productName;
  final int quantity;
  final double unitPrice;
  final double subtotal;

  InvoiceItem({
    required this.sku,
    required this.productName,
    required this.quantity,
    required this.unitPrice,
    required this.subtotal,
  });

  factory InvoiceItem.fromMap(Map<String, dynamic> m) => InvoiceItem(
        sku: m['sku'] ?? '',
        productName: m['product_name'] ?? '',
        quantity: (m['quantity'] ?? 1).toInt(),
        unitPrice: (m['unit_price'] ?? 0).toDouble(),
        subtotal: (m['subtotal'] ?? 0).toDouble(),
      );

  Map<String, dynamic> toMap() => {
        'sku': sku,
        'product_name': productName,
        'quantity': quantity,
        'unit_price': unitPrice,
        'subtotal': subtotal,
      };
}

class Invoice {
  final String invoiceId;
  final String customer;
  final String customerPhone;
  final String customerAddress;
  final String customerRfc;
  final List<InvoiceItem> items;
  final double total;
  final String channel;
  final String timestamp;
  final String profileId;
  final String notes;
  final String? pdfUrl;

  Invoice({
    required this.invoiceId,
    required this.customer,
    this.customerPhone = '',
    this.customerAddress = '',
    this.customerRfc = '',
    required this.items,
    required this.total,
    this.channel = 'manual',
    required this.timestamp,
    required this.profileId,
    this.notes = '',
    this.pdfUrl,
  });

  factory Invoice.fromMap(Map<String, dynamic> m, String profileId) => Invoice(
        invoiceId: m['invoice_id'] ?? '',
        customer: m['customer'] ?? '',
        customerPhone: m['customer_phone'] ?? '',
        customerAddress: m['customer_address'] ?? '',
        customerRfc: m['customer_rfc'] ?? '',
        items: (m['items'] as List? ?? [])
            .map((i) => InvoiceItem.fromMap(i))
            .toList(),
        total: (m['total'] ?? 0).toDouble(),
        channel: m['channel'] ?? 'manual',
        timestamp: m['timestamp'] ?? '',
        profileId: profileId,
        notes: m['notes'] ?? '',
        pdfUrl: m['pdf_url'] as String?,
      );

  Map<String, dynamic> toMap() => {
        'invoice_id': invoiceId,
        'customer': customer,
        'customer_phone': customerPhone,
        'customer_address': customerAddress,
        'customer_rfc': customerRfc,
        'items': items.map((i) => i.toMap()).toList(),
        'total': total,
        'channel': channel,
        'timestamp': timestamp,
        'profile_id': profileId,
        'notes': notes,
      };
}

File: desktop_java/src/main/java/com/mfagent/model/Invoice.java
package com.mfagent.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

/**
 * Equivalente al dict de factura/venta en Python.
 */
public class Invoice {
    @SerializedName("invoice_id")    private String invoiceId;
    private String timestamp;
    private String customer;
    @SerializedName("customer_phone")   private String customerPhone;
    @SerializedName("customer_address") private String customerAddress;
    @SerializedName("customer_rfc")     private String customerRfc;
    private String notes;
    private String channel;
    @SerializedName("registered_by")  private String registeredBy;
    @SerializedName("business_name")  private String businessName;
    private double total;
    private List<InvoiceItem> items = new ArrayList<>();

    public Invoice() {}

    // Getters y setters
    public String getInvoiceId()                    { return invoiceId; }
    public void setInvoiceId(String invoiceId)      { this.invoiceId = invoiceId; }

    public String getTimestamp()                    { return timestamp; }
    public void setTimestamp(String timestamp)      { this.timestamp = timestamp; }

    public String getCustomer()                     { return customer; }
    public void setCustomer(String customer)        { this.customer = customer; }

    public String getCustomerPhone()                        { return customerPhone; }
    public void setCustomerPhone(String customerPhone)      { this.customerPhone = customerPhone; }

    public String getCustomerAddress()                          { return customerAddress; }
    public void setCustomerAddress(String customerAddress)      { this.customerAddress = customerAddress; }

    public String getCustomerRfc()                  { return customerRfc; }
    public void setCustomerRfc(String customerRfc)  { this.customerRfc = customerRfc; }

    public String getNotes()                { return notes; }
    public void setNotes(String notes)      { this.notes = notes; }

    public String getChannel()              { return channel; }
    public void setChannel(String channel)  { this.channel = channel; }

    public String getRegisteredBy()                     { return registeredBy; }
    public void setRegisteredBy(String registeredBy)    { this.registeredBy = registeredBy; }

    public String getBusinessName()                     { return businessName; }
    public void setBusinessName(String businessName)    { this.businessName = businessName; }

    public double getTotal()                { return total; }
    public void setTotal(double total)      { this.total = total; }

    public List<InvoiceItem> getItems()             { return items; }
    public void setItems(List<InvoiceItem> items)   { this.items = items; }

    public void recalcTotal() {
        this.total = items.stream().mapToDouble(InvoiceItem::getSubtotal).sum();
    }
}
